### PR TITLE
Readability improvements

### PR DESCRIPTION
**Pull request by hackerbirds**
See #3 for more discussion

This PR doesn't attempt to modify the logic of the program whatsoever, so it should "just work" as it is, but I wasn't able to do any integration tests on it because I don't know how to for now
on
- Use format!() macro to build Strings to prevent appending String + &str
- Some print statements could be simplified by moving the variables into the string literal. This only works for types who implement `Display` though, so it can't be done everywhere. `{foo:?}` works for `Debug` types though, so some extra simplification could be done
- Mainly avoid using "&*" on String/str
- Consume iterators where possible
- Use a `const` value for the Dbus interface name/path where possible

### CODE REVIEW

File: src/bin/notification-proxy-client.rs [hunk 1]
```diff
@@ -160,24 +156,23 @@ impl Server {
                     })
                 }
                 "sound-file" => {
-                    eprintln!("Not yet implemented: Sound files (got {:?})", j)
+                    eprintln!("Not yet implemented: Sound files (got {j})")
```
Comment (DemiMarie) [inline src/bin/notification-proxy-client.rs]: I’m actually surprised that this implements `Display`.  I would prefer `Debug`, though, unless the `Display` implementation escapes special characters.

That said, even better might be to just drop the `j` entirely.

File: src/bin/notification-proxy-client.rs [hunk 2]
```diff
@@ -160,24 +156,23 @@ impl Server {
...
+                }
+                "sound-name" => {
+                    eprintln!("Not yet implemented: Sound files specified by name (got {j})")
```
Comment (DemiMarie) [inline src/bin/notification-proxy-client.rs]: Ditto.

File: src/bin/notification-proxy-client.rs [hunk 3]
```diff
@@ -160,24 +156,23 @@ impl Server {
...
                 }
-                "sound-name" => eprintln!(
-                    "Not yet implemented: Sound files specified by name (got {:?})",
-                    j
-                ),
                 "suppress-sound" => suppress_sound = true,
                 "transient" => transient = true,
                 "resident" => resident = true,
-                "x" | "y" => eprintln!("Ignoring coordinate hint {} {:?}", i, j),
+                "x" | "y" => eprintln!("Ignoring coordinate hint {i} {j}"),
```
Comment (DemiMarie) [inline src/bin/notification-proxy-client.rs]: This should check that `j` is an integer before calling `Display::display`.

File: src/bin/notification-proxy-client.rs [hunk 4]
```diff
@@ -160,24 +156,23 @@ impl Server {
...
                 "urgency" => match j {
                     Value::U8(0) => urgency = Some(Urgency::Low),
                     Value::U8(1) => urgency = Some(Urgency::Normal),
                     Value::U8(2) => urgency = Some(Urgency::Critical),
-                    _ => eprintln!("Ignoring unknown urgency value {:?}", j),
+                    _ => eprintln!("Ignoring unknown urgency value {j}"),
```
Comment (DemiMarie) [inline src/bin/notification-proxy-client.rs]: Again, there needs to be a check that the type of `j` is correct.

**Review by DemiMarie — COMMENTED** [review]
See comments.

File: src/bin/notification-proxy-client.rs (hunk 1 above)
Comment (hackerbirds) [reply src/bin/notification-proxy-client.rs]: You're right, this was a careless change, I will change `{j}` to `{j:?} to keep the prints how they originally were.

**Comment by hackerbirds** [comment]
It looks like `j` implements `Display`, but I will print `j` as `Debug` to keep the formatting as it was.

**Pull request closed without merging**